### ISSUE TITLE
"Possibly dangling reference" warning building clienttoolmodel.cpp

### ISSUE THREAD

**Issue by ferdnyc** [KDAB/GammaRay#871]
When building `ui/clienttoolmodel.cpp` in the 3.0.0 release code, GCC 13.2.1 is showing these warnings (line-wrapped for readability):

```text
/builddir/build/BUILD/gammaray-3.0.0/ui/clienttoolmodel.cpp: 
In member function 'virtual QVariant 
GammaRay::ClientToolModel::data(const QModelIndex&, int) const':
/builddir/build/BUILD/gammaray-3.0.0/ui/clienttoolmodel.cpp:42:21: 
warning: possibly dangling reference to a temporary [-Wdangling-reference]
   42 |     const ToolInfo &tool = m_toolManager->tools().at(index.row());
      |                     ^~~~
/builddir/build/BUILD/gammaray-3.0.0/ui/clienttoolmodel.cpp:42:53: 
note: the temporary was destroyed at the end of the full expression 
'GammaRay::ClientToolManager::tools() const().QList<GammaRay::ToolInfo>::at(
((qsizetype)(& index)->QModelIndex::row()))'
   42 |     const ToolInfo &tool = m_toolManager->tools().at(index.row());
      |                            ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
/builddir/build/BUILD/gammaray-3.0.0/ui/clienttoolmodel.cpp: 
In member function 'virtual Qt::ItemFlags 
GammaRay::ClientToolModel::flags(const QModelIndex&) const':
/builddir/build/BUILD/gammaray-3.0.0/ui/clienttoolmodel.cpp:92:17: 
warning: possibly dangling reference to a temporary [-Wdangling-reference]
   92 |     const auto &tool = m_toolManager->tools().at(index.row());
      |                 ^~~~
/builddir/build/BUILD/gammaray-3.0.0/ui/clienttoolmodel.cpp:92:49: 
note: the temporary was destroyed at the end of the full expression 
'GammaRay::ClientToolManager::tools() const().QList<GammaRay::ToolInfo>::at(
((qsizetype)(& index)->QModelIndex::row()))'
   92 |     const auto &tool = m_toolManager->tools().at(index.row());
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
```

After reading the code it _doesn't_ seem to me as if there's a temporary being created, or an actual dangling reference. But it's weird that GCC seems to _think_ there is.

Is there any actual issue here? And if not, can the code be restructured in a way that doesn't confuse GCC?

**Comment by Waqar144** [comment]
I guess doing `const auto &tools = m_toolManager->tools()` and then accessing `tools` would avoid the warning.

**Comment by ferdnyc** [comment]
@Waqar144 :+1: I'll give that a try, and submit a PR if it works out.

**Comment by ferdnyc** [comment]
Oh, never mind, it seems this was already addressed in 4ecfa803c by making them non-reference variables.
